packages/test/test-end-to-end-tests/src/test/gc/gcTreeSummaryHandles.spec.ts

```diff
@@ -151,19 +283,20 @@ describeNoCompat("GC Tree stored as a handle in summaries", (getTestObjectProvid
             );
 
             mainContainer = await createContainer();
-            dataStoreA = await requestFluidObject<TestDataObject>(mainContainer, "default");
-
-            summarizerClient1 = await getNewSummarizer();
+            dataStoreA = await requestFluidObject<ITestFluidObject>(mainContainer, "default");
 
             // Create data stores B and C, and mark them as referenced.
-            dataStoreB = await dataObjectFactory.createInstance(dataStoreA.containerRuntime);
-            dataStoreA._root.set("dataStoreB", dataStoreB.handle);
-            dataStoreC = await dataObjectFactory.createInstance(dataStoreA.containerRuntime);
-            dataStoreA._root.set("dataStoreC", dataStoreC.handle);
+            dataStoreB = await requestFluidObject<ITestFluidObject>(
```

Comment:
I'm curious as to what the difference between `dataObjectFactory.createInstance` and `containerRuntime.createDataStore` is and which one is better to use in the tests?

Reply:
In general, factory.createInstance is preferable because its type safe since you know exactly what the object you are getting back is. createDataStore returns a generic object and you need to know what the return type is.
In tests, it doesn't really matter that much because its a controlled environment. However, the reason I am not using the factory here is because the test doesn't have access to it.
Once this bug is fixed, we could have access to it and called createInstance on it - https://dev.azure.com/fluidframework/internal/_workitems/edit/756